hyperspy/_signals/eds.py

```diff
@@ -617,6 +617,11 @@ def get_lines_intensity(self,
 
         """
 
+        if not (xray_lines is None or isinstance(xray_lines, list)):
```

Comment:
A further improvement is to support any compatible iterable type, such as numpy array and tuple, which could be done with:
```python
from collections.abc import Iterable

if isinstance(xray_lines, Iterable) and not isinstance(xray_lines, str):
   ...
```
since str object are also iterable. In this case, it would be good to update the docstring too `{None, list of string}` -> `{None, iterable of string}`.

Reply:
I incorporated your suggestion, with a small caveat. I'll add a comment to the new commit.